Truncate cy.log() in Command Log to 50 lines instead of 1

pitgrap opened the pull request:
- Closes #5625 

### User facing changelog

Messages from `cy.log()` are now truncated at 50 lines the Command Log instead of the 1 line.

### How has the user experience changed?

Before:
![screenshot_174](https://user-images.githubusercontent.com/227391/71263699-b766c200-2342-11ea-85bb-419df78ca0a6.png)

After:

Truncates the log to 50 lines and shows ellipsis to indicate truncation.

Clicking to log to console will log full `cy.log()` content. 

<img width="517" alt="Screen Shot 2020-01-09 at 12 40 53 PM" src="https://user-images.githubusercontent.com/1271364/72042581-55fc9800-32dd-11ea-908e-6f4a29dce5ea.png">



### PR Tasks

none.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress?pullRequest=5630) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress?pullRequest=5630) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress?pullRequest=5630) it.</sub>

brian-mann commented:
@jennifer-shehane shouldn't there be some kind of upper maximum here? Seems slightly irresponsible to display **all** of the log message regardless of how long it is. 

With this PR, if you shrink the reporter, the text will wrap really far and make the command log pretty unusable at smaller sizes. Could we truncate the log message to a more reasonable point so we avoid that problem?

pitgrap commented:
@brian-mann As a cypress test code writer, I would like to have control of the log messages of my own tests. This change does not affect any other message types. Just my own written `cy.log()`. I don't know why cypress should restrict me as developer? Also the Command Log has a minimum width of 170px.
But if I still couldn't convince you, a simple `max-height: 300px;` should do it, because the default overflow/text-overflow rules would apply.

jennifer-shehane commented:
@pitgrap The restriction would be a very large one, like 2,000 characters. The purpose of restriction would be to prevent the reporter from printing a gigantic log, like if someone accidentally printed a large Object, that may lock up the browser during rendering. Something more than this amount of characters wouldn't even be visible in a single frame of the test runner for review. Users can open a new issue if they need this to be larger and handled in some other way.

I'll update the PR for this.

chrisbreiding commented on packages/reporter/src/commands/commands.scss:
```diff
@@ -377,6 +377,13 @@
+    .command-message {
+      white-space: initial;
+      word-wrap: break-word;
+    }
```
Let's add `line-height: 1.5` to space out the lines a bit more.

pitgrap replied:
Done.

pitgrap commented:
Do you have any updates? Can I do anything to get this PR merged?

gabbersepp commented:
Hello, can this PR be merged? Please have a look at this screenshot, taken from one of our testruns:
![image](https://user-images.githubusercontent.com/3234900/70897285-97f23100-1ff2-11ea-9207-8f27229050dc.png)

I have no chance to check if the lcoator and the element in the UI are correct. My assumption is, that the element was replaced dynamically but as I can not see the full locator of the `get` I have no chance to confirm this.

jennifer-shehane commented:
There were changes requested for this PR that were never implemented. Implement the changes and we can move forward. 

- Set upper limit for logs display
- https://github.com/cypress-io/cypress/pull/5630#pullrequestreview-314373999

Likely relevant to this work https://github.com/cypress-io/cypress/pull/5762

gabbersepp commented:
@jennifer-shehane Thanks for pointing that out.
@pitgrap Can you have a look again? Would be nice!

pitgrap commented:
Rebased on develop branch, added the feedback and updated the screenshots.
Used a simple CSS solution for the upper limit based on https://caniuse.com/#feat=css-line-clamp

pitgrap commented:
> If you are going to truncate, we also need some ellipsis to indicate that the log was truncated otherwise the users will have no way of knowing whether it is the end of the log or truncated.

I agree, and that's what should happen. See my screenshot in the description. 


> I do not see the `line-clamp` style have any effect. As can be seen in the image below, the log is printing 50+ lines, not truncating at line 50.

That's weird. I will retest it.

pitgrap commented:
Ok, sorry. My new styles has been overwritten by another CSS rule with a "display: block". Now it should be finally fixed and working.
